fs/artifact_fetcher.go

```diff
@@ -66,32 +71,75 @@ func newArtifactFetcher(refspec reference.Spec, localStore, remoteStore content.
 	}, nil
 }
 
+// All three (jitter,backoffStrategy,retryStrategy) are copied over from
+// https://github.com/awslabs/soci-snapshotter/blob/main/fs/remote/resolver.go.
+// TODO: Consolidate retry/http logic into its own package.
+func jitter(duration time.Duration, divisor int64) time.Duration {
+	return time.Duration(rand.Int63n(int64(duration)/divisor) + int64(duration))
+}
+
+func backoffStrategy(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+	delayTime := rhttp.DefaultBackoff(min, max, attemptNum, resp)
+	return jitter(delayTime, 8)
+}
+
+func retryStrategy(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	retry, err2 := rhttp.DefaultRetryPolicy(ctx, resp, err)
+	if retry {
+		log.G(ctx).WithFields(logrus.Fields{
+			"error":    err,
+			"response": resp,
+		}).Debugf("fetching artifacts: Retrying request")
+	}
+	return retry, err2
+}
+
 func newRemoteStore(refspec reference.Spec) (*remote.Repository, error) {
 	repo, err := remote.NewRepository(refspec.Locator)
 	if err != nil {
 		return nil, fmt.Errorf("cannot create repository %s: %w", refspec.Locator, err)
 	}
 
-	authClient := auth.DefaultClient
-	authClient.Cache = auth.DefaultCache
-	authClient.Credential = func(_ context.Context, host string) (auth.Credential, error) {
-		username, secret, err := dockerconfig.DockerCreds(host)
-		if err != nil {
-			return auth.EmptyCredential, err
-		}
-		if username == "" && secret != "" {
+	const defaultTimeout = 2 * time.Second
```

Comment:
can we move this const to the beginning of the file and add some comment about the value / what it is used for.